community-events/content/challenge-project/03-exercise-portal-deploy.md

```diff
@@ -0,0 +1,162 @@
+
+```bash
+  cd packages/portal-deploy
+```
+
+We need to delete the `packages/portal/node_modules` folder and the `swa-cli.config.json` file. To do this, we will use the linux command `rm -rf`. The -r is used to delete folders and the -f is used to force deletion.
+
+**Remove the node_modules folder:**
+
+```bash
+  rm -rf node_modules
+```
+
+**Remove the swa-cli.config.json file:**
+
+```bash
+  rm -rf swa-cli.config.json
+```
+
+### Installing dependencies
+
+**All depedencies**
+
+```bash
+  npm install
+```
+
+**Azure Static Web Apps CLI as a development dependency**
+
+```bash
+  npm install -D @azure/static-web-apps-cli
+```
+
+## Azure Static Web Apps (SWA) CLI
+
+With the initial settings completed, you will need to initialize Azure Static Web Apps so that we can deploy the `portal` to Azure.
+
+To do this, use the terminal to navigate to the root of the Contoso Real Estate project by running the following command:
+
+```bash
+  cd ../..
+```
+
+**Initialize Azure Static Web Apps**
+
+```bash
+  swa init
+```
+
+The SWA CLI will identify the existing projects in Contoso Real Estate and you should choose the `portal-deploy`. Check if the project name is correct and press `y` to confirm.
+
+![Visual Studio Code Terminal in Codespaces](./images/swa-init.gif)
+
+A `swa-cli.config.json` file will be generated and will have the following information:
+
+```json
+  {
+    "$schema": "https://aka.ms/azure/static-web-apps-cli/schema",
+    "configurations": {
+      "contoso-real-estate": {
+        "appLocation": "packages/portal-deploy",
+        "apiLocation": "packages/api",
+        "outputLocation": "dist/contoso-app",
+        "apiLanguage": "node",
+        "apiVersion": "16",
+        "appBuildCommand": "npm run build",
+        "apiBuildCommand": "npm run build --if-present",
+        "run": "npm start",
+        "appDevserverUrl": "http://localhost:4200"
+      }
+    }
+  }
+```
+
+Note that the `appLocation` property is pointing to the `portal-deploy` folder, which is the project we will deploy.
+
+**Execution test**
+
+```bash
+  swa start
+```
+
+During execution, a popup will open in the lower right corner of Visual Studio Code. Click on the `Ports` tab and check if port `4280` is open. If it is, click the `Open Browser` button so you can view the portal.
+
+![Ports](./images/ports-executing-new-deploy.gif)
+
+
+### Creating the artifacts
+
+Artefacts are the files that will be sent to Azure Static Web Apps. Artifacts are static files, such as HTML, CSS, JavaScript, and images, that are generated by the build process. All artefacts generated by the SWA CLI are stored in the `dist/contoso-app` folder. To generate the `portal` artefacts, run the following command:
+
+```bash
+  swa build
+```
+
+### Configuring GitHub Actions
+
+Now that we already have the generated artefacts, we need to configure GitHub Actions so that the `portal-deploy` is published on Azure Static Web Apps.
```

Comment:
```suggestion
Now that we already have the generated artifacts, we need to configure GitHub Actions so that the `portal-deploy` is published on Azure Static Web Apps.
```